tests/arch/arc/arc_dsp_sharing/src/dsp_regs_arc.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file
+ * @brief ARC specific dsp register macros
+ */
+
+/*
+ * Copyright (c) 2022, Synopsys.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _DSP_REGS_ARC_H
+#define _DSP_REGS_ARC_H
+
+#include <zephyr/toolchain.h>
+#include "dsp_context.h"
+
+/**
+ *
+ * @brief Load all dsp registers
+ *
+ * This function loads all DSP and AGU registers pointed to by @a regs.
+ * It is expected that a subsequent call to _store_all_dsp_registers()
+ * will be issued to dump the dsp registers to memory.
+ *
+ * The format/organization of 'struct dsp_register_set'; the generic C test
+ * code (main.c) merely treat the register set as an array of bytes.
+ *
+ * The only requirement is that the arch specific implementations of
+ * _load_all_dsp_registers() and _store_all_dsp_registers() agree
+ * on the format.
+ *
+ */
+static inline void _load_all_dsp_registers(struct dsp_register_set *regs)
+{
+	uint32_t temp = 0;
+
+	__asm__ volatile (
+		"ld %1, [%0, 0];\n\t"
+		"sr %1, [%2];\n\t"
+		"ld %1, [%0, 4];\n\t"
+		"sr %1, [%3];\n\t"
+		"ld %1, [%0, 8];\n\t"
+		"sr %1, [%4];\n\t"
+		: : "r" (regs), "r" (temp),
+		"i"(_ARC_V2_DSP_BFLY0), "i"(_ARC_V2_AGU_AP0),
+		"i"(_ARC_V2_AGU_OS0)
+		: "memory"
+		);
```

Comment:
Hmm, I'm wondering if we actually need inline ASM in _load_all_dsp_registers / _store_all_dsp_registers or we can simply replace it ith aux_read / aux write?

Reply:
Instructions related to aux can be replaced with aux_read / aux write, but we still need `st` and `ld` to write and read from stack.

Reply:
> Instructions related to aux can be replaced with aux_read / aux write, but we still need st and ld to write and read from stack.

As I can see we are not reading / writing from stack - we are reading from dsp_register_set structure. So we can just read  / write from/to member of this structure. I don't see any reason to do it in ASM.

Please correct me if I'm wrong.